File: All_Topics/Valid_Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        
        int len = s.size() - 1;
        stack<char> st;

        for(int i = 0; i <= len; i++)
        {
            if(s[i] == '(' || s[i] == '{' || s[i] == '[')
                st.push(s[i]);
            else if(s[i] == ')' || s[i] == '}' || s[i] == ']')
            {
                if(st.empty())
                    return false;
                else if(s[i] == ')' && st.top() != '(')
                    return false;
                else if(s[i] == '}' && st.top() != '{')
                    return false;
                else if(s[i] == ']' && st.top() != '[')
                    return false;
                else
                    st.pop();                
            }
        }
        return st.empty();
    }
};

File: Blind_75_LeetCode_Questions/java/String/Q20_Valid_Parentheses/Valid_Parentheses.java
class Solution {
    public boolean isValid(String s) 
    {
        Stack<Character> stack = new Stack<>();
        
        for(char c : s.toCharArray())
        {
            if(c == '(' || c == '{' || c == '[')
                stack.push(c);
            else
            {
                if(stack.isEmpty())
                    return false;
                char ch = stack.pop();

                if(c == ')' && ch != '(' || c == '}' && ch != '{' || c == ']' && ch != '[')
                    return false;
            }
        }
        return stack.isEmpty();
    }
}
